src/main/py/peer/peer.py

```
@@ -156,118 +158,186 @@ async def commission_art_piece(self) -> None:
+                exchange_key, "", announcement.get_price(), self.keys["public"]
+            )
         )
 
         response_key = utils.generate_random_sha1_hash()
 
-        self.inventory.add_pending_trade(
-            trade_key,
-            offer_response,
-        )
+        if announcement.get_exchange_type() == "trade":
+            self.inventory.add_pending_exchange(
+                exchange_key,
+                offer_response,
+            )
         try:
             set_success = await self.node.set(
                 response_key, pickle.dumps(offer_response)
             )
             if set_success:
-                self.logger.info("Trade response sent")
+                self.logger.info("%s response sent", announcement.get_exchange_type())
             else:
-                self.logger.error("Trade response failed to send")
+                self.logger.error(
+                    "%s response failed to send", announcement.get_exchange_type()
+                )
+
         except TypeError:
-            self.logger.error("Trade response type is not pickleable")
+            self.logger.error(
+                "%s response type is not pickleable", announcement.get_exchange_type()
+            )
 
-    async def handle_accept_trade(self, response: OfferResponse):
-        """Handle an accepted trade"""
+    async def handle_exchange_response(
+        self,
+        exchange_key: bytes,
+        response: OfferResponse,
+    ):
+        """
+        Handle an exchange response from the network.
+        """
 
-        self.logger.info(response)
+        self.logger.info("Handling exchange response")
+        if exchange_key in self.inventory.pending_exchanges:
+            self.inventory.remove_pending_exchange(exchange_key)
+            if response.get_price() == 0:
+                if response.exchange_id in self.inventory.pending_exchanges:
+                    self.inventory.remove_pending_exchange(response.trade_id)
```

`response.exchange_id`
maybe a missing test, since this should raise an exception if this line is executed